backend/capellacollab/sessions/operators/k8s.py

```diff
@@ -185,23 +257,41 @@ def _start_session(
             image=image,
             name=_id,
             environment=environment,
+            ports=ports,
             persistent_workspace_claim_name=persistent_workspace_claim_name,
             pure_variants_secret_name=pure_variants_secret_name,
+            limits=limits,
         )
 
-        service = self._create_service(name=_id, deployment_name=_id)
+        service = self._create_service(
+            name=_id,
+            deployment_name=_id,
+            ports=ports,
+            prometheus_path=prometheus_path,
+            prometheus_port=prometheus_port,
+        )
 
         log.info(
             "Launched a %s session for user %s with id %s",
             session_type,
             username,
             _id,
         )
-        return self._export_attrs(deployment, service)
+        return self._export_attrs(deployment, service, ports)
 
     def kill_session(self, _id: str):
         log.info("Terminating session %s", _id)
 
+        if self.openshift:
+            if dep_status := self._delete_openshift_route(_id):
+                log.info(
+                    "Deleted route %s with status %s", _id, dep_status.status
+                )
+        if dep_status := self._delete_ingress(_id):
+            log.info(
+                "Deleted ingress %s with status %s", _id, dep_status.status
+            )
+
```

Comment:
```suggestion
        if self.openshift:
            if dep_status := self._delete_openshift_route(_id):
                log.info(
                    "Deleted route %s with status %s", _id, dep_status.status
                )
        else:
            if dep_status := self._delete_ingress(_id):
                log.info(
                    "Deleted ingress %s with status %s", _id, dep_status.status
                )
```
Without this, we would always produce an error on openshift